homeassistant/core.py

```diff
@@ -1257,6 +1265,100 @@ def as_dict(self) -> Dict:
             'config_source': self.config_source
         }
 
+    def set_time_zone(self, time_zone_str: Optional[str]) -> None:
+        """Help to set the time zone."""
+        if time_zone_str is None:
+            return
+
+        time_zone = dt_util.get_time_zone(time_zone_str)
+
+        if time_zone:
+            self.time_zone = time_zone
+            dt_util.set_default_time_zone(time_zone)
+        else:
+            _LOGGER.error("Received invalid time zone %s", time_zone_str)
+
+    @callback
+    def _update(self,
+                source: str,
```

Comment:
Can you add `*,` before source , so that we only accept keyword arguments

Reply:
Fixed.